src/components/panel/panel.js

```diff
@@ -1563,22 +1624,161 @@ MdPanelAnimation.prototype.closeTo = function(closeTo) {
  * Returns the element and bounds for the animation target.
  * @param {string|!Element|{top: number, left: number}} location
  * @returns {!{element: !angular.JQLite|undefined, bounds: !DOMRect}}
+ * @private
  */
-MdPanelAnimation.prototype.getPanelAnimationTarget = function(location) {
+MdPanelAnimation.prototype._getPanelAnimationTarget = function(location) {
   if (angular.isDefined(location.top) || angular.isDefined(location.left)) {
     return {
       element: undefined,
       bounds: {
         top: location.top || 0,
-        left: location.left || 0,
-        height: 1,
-        width: 1
+        left: location.left || 0
       }
     };
   } else {
     return this._getBoundingClientRect(this._getElement(location));
   }
-}
+};
+
+
+/**
+ * Specifies the animation class.
+ *
+ * There are several default animations that can be used:
+ *   `md-panel-animate-slide`: The panel slides in and out from the specified
+ *        elements.
+ *   `md-panel-animate-scale`: The panel scales in and out.
+ *   `md-panel-animate-fade`: The panel fades in and out.
+ *
+ * @param {string|!{open: string, close: string}} cssClass
+ * @returns {MdPanelAnimation}
+ */
+
+MdPanelAnimation.prototype.withAnimation = function(cssClass) {
```

Comment:
What do you think about the built-in animations being available as something like
`.withAnimation($mdPanel.animation.Slide)`

This way the API isn't tightly coupled to always being just one css class; I might want to later add, for example, an animation that flies a few loops around the page before landing.